src/main/webapp/user-private-comments.jsp

```diff
@@ -56,12 +56,9 @@
             <form action="/user-profile" method="POST">
               <input type="hidden" name="taskID" value="<%=task.getTaskID()%>"> 
               <br>
-              <%
-                String ID = (String) "box-" + task.getTaskID();
-              %>
-              <div id="box-<%=task.getTaskID()%>" onload="init('<%=ID%>')" type="text"></div>
-              <input type="hidden" name="message-<%=task.getTaskID()%>" value="document.getElementById('<%=ID%>').value(outputData);"</input>
-              <button onclick="save()" class="deep-button submit-button" type="submit" class="deep-button">Update Comment</button>
+              <div id="box-<%=task.getTaskID()%>" onload="init(<%=task.getTaskID()%>)" type="text">Here</div>
+              <button onclick="save(init(<%=task.getTaskID()%>))" class="deep-button submit-button" type="submit" class="deep-button">Update Comment</button>
```

Comment:
Pass init(taskID) which returns editor to save().

Reply:
calling init() again here would probably reinitialize the box (would reset to the beginning state), so you'd loose any edits you've done in the data.

Alternative approach would be have init save the editor in a map, then save would lookup the editor from the map:

```
var editors = {}
function init(pcID) {
 // do stuff to init box
  editors[pcID] = editor;
}

function save(pcID) {
  editor = editors[pcID];
  // do save stuff to editor
}
```